aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -197,10 +197,39 @@ pub(crate) fn get_or_vm_startup_failure<'a, T>(
     })
 }
 
+/// Checks if a given transaction is a governance proposal by checking if it has one of the
+/// approved execution hashes.
+fn is_governance_proposal<R, E>(
+    resolver: &R,
+    txn: &SignedTransaction,
+    txn_metadata: &TransactionMetadata,
+) -> bool
+where
+    R: MoveResolver<E>,
+    E: Debug,
+{
+    match txn.payload() {
+        TransactionPayload::Script(_script) => {
+            match resolver.get_resource(&CORE_CODE_ADDRESS, &ApprovedExecutionHashes::struct_tag())
```

Comment:
`ApprovedExecutionHashes` implements `OnChainConfig`, let's not use MoveResolver in this case and deserialization explicitly. I think you can do:

```rust
fn is_governance_proposal(
    config_storage: &impl ConfigStorage,
    txn: &SignedTransaction,
    txn_metadata: &TransactionMetadata,
) -> bool {
  ...
  match ApprovedExecutionHashes::fetch_config(config_storage) {
    Some(...) => ...
    None => ...
  }
  ...
}
```